scripts/ec2/start_instance.coffee

```diff
@@ -8,13 +8,38 @@
 #   instance_id : [required] The ID of one or more instances to tag. For example, i-0acec691.
 #   --dry-run  : [optional] Checks whether the api request is right. Recommend to set before applying to real asset.
 
+ec2 = require('../../ec2.coffee')
+restrictor = require './restrictor'
 util = require 'util'
 
 getArgParams = (arg) ->
   dry_run = if arg.match(/--dry-run/) then true else false
-
   return {dry_run: dry_run}
 
+
+startInstances = (msg, params, instances, err) ->
+  return (err) ->
+    if err
+      msg.send "Error! #{err}"
+      return
+
+    dry_run = params.dry_run
+    msg.send "Starting instances=[#{instances}] dry-run=#{dry_run}..."
+
+    params = {InstanceIds: instances, DryRun: dry_run}
+
+    if dry_run
+      msg.send util.inspect(params, false, null)
+
+    ec2.startInstances params, (err, res) ->
+      if err
+        msg.send "Error: #{err}"
+      else
+       msg.send "Success! The instances are starting"
+       msg.send util.inspect(res, false, null)
```

Comment:
In the same vein it's a bit more idiomatic to use the return-early pattern when possible:

```
return msg.send "Error: #{err}" if err

msg.send "Success! The instances are starting"
msg.send util.inspect(res, false, null)
```